File: mo-backend/mo-boot-seed/src/main/java/me/puras/mo/seed/moboot/service/user/UserServiceImpl.java
package me.puras.mo.seed.moboot.service.user;

import me.puras.mo.seed.moboot.domain.User;
import me.puras.common.persistence.CrudRepository;
import me.puras.common.service.CrudServiceImpl;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * @author <a href="mailto:he@puras.me">puras</a>
 * @version $Revision 1.0 $
 * Created On 2017-08-02 11:23
 */
@Service
@Transactional
public class UserServiceImpl extends CrudServiceImpl<User> implements UserService {
    @Autowired
    private UserRepository repository;

    @Override
    public CrudRepository<User> getRepository() {
        return repository;
    }
}

File: mo-backend/mo-kotlin-seed/src/main/kotlin/me/puras/mo/seed/kotlin/service/user/UserServiceImpl.kt
package me.puras.mo.seed.kotlin.service.user

import me.puras.common.persistence.CrudRepository
import me.puras.common.service.CrudServiceImpl
import me.puras.mo.seed.kotlin.domain.User
import org.springframework.stereotype.Service

@Service
class UserServiceImpl(var repository: UserRepository) : CrudServiceImpl<User>(), UserService {
    override fun getRepository(): CrudRepository<User> {
        return repository
    }
}